crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -1124,7 +1125,57 @@ mod tests {
             });
         });
 
-        assert_eq!(entity.get::<TestComponent>().unwrap().0, 42);
-        assert_eq!(entity.get::<TestComponent2>().unwrap().0, 84);
+        let dynamic_res: Vec<_> = world
+            .query::<(&TestComponent, &TestComponent2)>()
+            .iter(&world)
+            .collect();
+
+        assert_eq!(dynamic_res, vec![(&TestComponent(42), &TestComponent2(84))]);
+
+        // Compare with `World` generated using static type equivalents
+        let mut static_world = World::new();
+
+        static_world.spawn((test_component_value, test_component_2_value));
+        let static_res: Vec<_> = static_world
```

Comment:
Nit: it's not clear to me what the `res` suffix means here.

Reply:
Done